app.js

```diff
@@ -62,6 +62,133 @@ VIEWER.isJSON = function(obj) {
     return r
 }
 
+
+VIEWER.navplaceObject = (function(geojson, navplaces, depth) {
+    //make sure value is a dict and not null
+    if (typeof geojson !== 'object' || geojson === null) {
+        return undefined;
+    }
+    //find navplace in the current object
+    if (geojson.hasOwnProperty('navPlace')) {
+        navplaces.push(geojson['navPlace'])
+    }
+    
+    if (depth > 5) {
+        console.warn("Maximum JSON search depth reached.")
+        return undefined;
+    }
+    //recursive
+    for (var key in geojson) {
+        var value = geojson[key];
+
+        if (Array.isArray(value)) {
+            for (var i = 0; i < value.length; i++) {
+                if (typeof value[i] === 'object' || Array.isArray(value[i])) {
+                    this.navplaceObject(value[i], navplaces, depth+1);
+                }
+            }
+        } else if (typeof value === 'object') {
+            this.navplaceObject(value, navplaces, depth+1);
+        }
+    }
+    return navplaces.filter(item => item !== undefined);
+});
+
+VIEWER.getBbox = (function(navplaceObj){    
```

Comment:
I will not claim that I understand this exactly line for line.  However, from what I understand, this is the kind of thing it needed to do.  This was a good function to grab.